chat-plugins/mafia.js

```diff
@@ -1,919 +1,916 @@
+		this.lynches = {};
+		this.phase = 'day';
+		if (extention && !initial) {
+			// Day stays same
+			this.setDeadline(extention);
 		} else {
-			this.sendRoom('|uhtml|mafia' + this.game.room.gameNumber + 'vote' + this.game.gamestate + this.game.day + '|' + this.game.mafiaWindow(image, output));
+			this.dayNum++;
 		}
+		this.sendRoom(`Day ${this.dayNum}. The hammer count is set at ${this.getHammer()}`, {declare: true});
+		this.updatePlayers();
+		return true;
 	}
 
-	// Targeting mechanics:
-
-	// Targets a single player of side side.
-	singleTarget(side, targetSelf) {
-		this.validTargets = {};
-		for (let i in this.game.players) {
-			let thisSide = this.game.players[i].class.side;
-			if ((side === 'any' || thisSide === side || (side === 'nomafia' && thisSide !== 'mafia')) && (targetSelf || this.game.players[i] !== this)) {
-				this.validTargets[i] = this.game.players[i];
-			}
-		}
+	night(early) {
+		if (this.phase !== 'day') return false;
+		if (this.timer) this.setDeadline('off');
+		this.phase = 'night';
+		this.sendRoom(`Night ${this.dayNum}. PM the host your action, or idle.`, {declare: true});
+		if (!early && this.getPlurality()) this.sendRoom(`Plurality is on ${this.players[this.getPlurality()].name}`);
+		this.updatePlayers();
+		return true;
 	}
 
-	// Triggers after the user has selected their target.
-	onReceiveTarget(target) {
-		if (!this.targeting) {
-			return this.sendRoom("You're not selecting a target right now.");
-		}
-
-		if (target in this.validTargets || target === 'none') {
-			if (target === 'none') {
-				if (this.using) delete this.using;
-				this.toExecute = null;
-			} else {
-				this.target = this.game.players[target];
-			}
-
-			this.updateTarget(this.class.image);
-
-			for (let i in this.game.players) {
-				if (!this.game.players[i].done) {
-					return;
-				}
-			}
-			this.game.progress(true);
+	lynch(user, target) {
+		if (this.phase !== 'day') return false;
+		let player = this.players[user.userid];
+		if (!player && this.dead[user.userid] && this.dead[user.userid].restless) player = this.dead[user.userid];
+		if (!(target in this.players) && target !== 'nolynch') return false;
+		if (player.lynching || (target === player.userid && !this.selfEnabled)) return false;
+		let lynch = this.lynches[target];
+		if (!lynch) {
+			this.lynches[target] = {count: 1, lastLynch: Date.now(), dir: 'up', lynchers: [user.userid]};
+			lynch = this.lynches[target];
 		} else {
-			this.sendRoom("Invalid target");
-		}
+			lynch.count++;
+			lynch.lastLynch = Date.now();
+			lynch.dir = 'up';
+			lynch.lynchers.push(user.userid);
+		}
+		player.lynching = target;
+		if (this.getHammer() <= lynch.count) {
+			// HAMMER
+			this.sendRoom(`Hammer! ${target === 'nolynch' ? 'Nobody' : this.players[target].name} was lynched!`, {declare: true});
+			if (target !== 'nolynch') this.eliminate(target);
+			this.night(true);
+		}
+		// Ask about bandwidth concerns
+		//this.updatePlayers();
```

Comment:
What would this be sending and to who? Do they need to be sent this/see this update?

Reply:
All players and the host, updates the current lynch count without the need to use the refresh button I added to the HTML room.